include/bitboard.hpp

```diff
@@ -63,12 +63,26 @@ class Bitboard {
     // Bitwise NOT
     friend constexpr Bitboard operator~(const Bitboard& b) { return ~b; }
 
+    // Prints the bitboard in a 8x8 grid format ; 1 = piece, 0 = empty square
     friend std::ostream& operator<<(std::ostream& os, const Bitboard& b) {
-        // TODO: Fill in the function definition
 
+        for (int rank = 7; rank >= 0; rank--) {
+            for (int file = 0; file <= 7; file++) {
+                // rank * 8 + file gives the index of the square
+                if (b.board & (1ULL << (rank * 8 + file))) {
+                    os << "1 ";
+                } else {
+                    os << "0 ";
+                }
+            }
+            os << "\n";
+        }
         return os;
     }
 
+    // constructor with a default value of 0
+    constexpr Bitboard(std::uint64_t b = 0) : board(b) {}
+
```

Comment:
Put this at the top of the class definition, above all the operator overloads.

Reply:
done